src/memory/spi_mem.h

```diff
@@ -0,0 +1,80 @@
+// Copyright 2025 ShiftCrypto AG
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef __SPI_MEM_H
+#define __SPI_MEM_H
```

Comment:
super nit, but the rest of our code uses `_SPI_MEM_H_` for the header-style guard